packages/editor/src/diagram/activities/activity-views.tsx

```diff
@@ -14,28 +14,30 @@ const JSX = { createElement: svg };
 export class ActivityNodeView extends RectangularNodeView {
   @inject(CustomIconToggleActionHandler) @optional() protected customIconHandler?: CustomIconToggleActionHandler;
 
-  render(node: ActivityNode, context: RenderingContext): VNode {
+  render(node: ActivityNode, context: RenderingContext): VNode | undefined {
+    if (!this.isVisible(node, context)) {
+      return undefined;
+    }
     const rcr = this.getRoundedCornerRadius(node);
-    const width = Math.max(0, node.bounds.width);
-    const height = Math.max(0, node.bounds.height);
     return (
       <g>
         <rect
+          attrs={{ [ATTR_BBOX_ELEMENT]: true }}
```

Comment:
what is this used for?

Reply:
If that attribute is set, the bounds of this rect-element are being used to calculate the bounds of the element, e.g., ActivityNode. When we need to determine the sizes of the elements, they are rendered on the hidden graph of sprotty without any zooming etc. applied. During that [hidden rendering](https://github.com/eclipse-sprotty/sprotty/blob/8eb479fbd7b1836d32ad82d7a8188769a474598c/packages/sprotty/src/features/bounds/hidden-bounds-updater.ts#L158), we query the browser for the bounding box of the rendered element ([getBBox](https://developer.mozilla.org/en-US/docs/Web/API/SVGGraphicsElement/getBBox)) and apply that as size to the element. However, you can run into problems when the rendering of the element and the bounds of an element get in "conflict" with each other, e.g., when rendering children outside the bounds as decoration or - as I have learned - when rendering the diamond node which for some reason produces a different bounding box than expected. For this reason, the hidden bounds updater allows us to specify that attribute and if an element with that attribute is present, we are taking the bounds of that SVG element as the bounds for the GLSP element.